daemon/api.go

```diff
@@ -1132,7 +1132,13 @@ func snapUpdate(inst *snapInstruction, st *state.State) (string, []*state.TaskSe
 		return "", nil, err
 	}
 
-	ts, err := snapstateUpdate(st, inst.Snaps[0], inst.Channel, inst.Revision, inst.userID, flags)
+	ts, err := snapstateUpdate(st, inst.userID, &snapstate.UpdateOptions{
```

Comment:
it's probably indeed easier to do this in two parts, first the refactor then the new feature.  I think userID should still come toward the end of the signature.